pkg/web/oauthclient/cookie_state.go

```diff
@@ -44,7 +43,7 @@ func (oc *OAuthClient) StateCookie() *cookie.Cookie {
 		Name:     oc.config.StateCookieName,
 		HTTPOnly: true,
 		Path:     oc.getMountPath(),
-		MaxAge:   10 * time.Minute,
+		MaxAge:   0,
```

Comment:
I don't think this is a good idea. If you don't finish your login within a certain duration, we should not allow it anymore.

Reply:
Why?

You have to regard all of this in the context of our own internal OAuth clients that use the oauthclient package (such as the Console), for which we apply some special policies already since they are trusted by us. I don't think expiring the login is of any benefit and only introduces friction. I don't want users to be able to use the Console login only within 10 minutes of loading the page and I also don't want them to be consequently greeted by an error message telling them that the login failed because of some expiration that makes no sense to them (since for them logging into the OAuth provider and the Console is one and the same thing).

I get that for a normal OAuth flow that is initiated by a third party that we would apply different measures but not for our own internal clients.

Otherwise, the only way I see this implemented in a UX friendly way is the `oauthclient` package automatically redirecting to the Console root URL that would initiate the login flow again and directly log the user in, since the user is already logged into the Account App. I chose not to do this because it could result in redirect loops when the state cookie is not present for other reasons than expiry.